src/ParaTest/Coverage/CoverageMerger.php

```diff
@@ -19,26 +24,70 @@ public function addCoverage(\PHP_CodeCoverage $coverage)
     }
 
     /**
-     * @param \PHP_CodeCoverage $coverage
+     * @param \PHP_CodeCoverage|CodeCoverage $coverage
      */
-    private function setCoverage(\PHP_CodeCoverage $coverage)
+    private function setCoverage($coverage)
     {
         $this->coverage = unserialize(serialize($coverage));
     }
 
     /**
-     * @param \PHP_CodeCoverage $coverage
+     * @param \PHP_CodeCoverage|CodeCoverage $coverage
      */
-    private function mergeCoverage(\PHP_CodeCoverage $coverage)
+    private function mergeCoverage($coverage)
     {
         $this->coverage->merge($coverage);
     }
 
     /**
-     * @return \PHP_CodeCoverage
+     * Returns coverage object from file.
+     *
+     * @param string $coverageFile Coverage file.
+     *
+     * @return \PHP_CodeCoverage|CodeCoverage
+     */
+    private function getCoverageObject($coverageFile)
+    {
+        $coverage = file_get_contents($coverageFile);
+
+        if (substr($coverage, 0, 5) === '<?php') {
+            return include $coverageFile;
+        }
+
+        // the PHPUnit 3.x and below
+        return unserialize($coverage);
+    }
+
+    /**
+     * Adds the coverage contained in $coverageFile and deletes the file afterwards
+     * @param string $coverageFile
+     * @throws \RuntimeException When coverage file is empty
      */
-    public function getCoverage()
+    public function addCoverageFromFile($coverageFile)
     {
-        return $this->coverage;
+        if ($coverageFile === null || !file_exists($coverageFile)) {
+            return;
+        }
+
+        if (filesize($coverageFile) === 0) {
+            throw new \RuntimeException("Coverage file $coverageFile is empty. This means a PHPUnit process has crashed.");
+        }
+
+        $this->addCoverage($this->getCoverageObject($coverageFile));
+        unlink($coverageFile);
+    }
+
+    /**
+     * Get coverage report generator
+     *
+     * @return CoverageReporterInterface
+     */
+    public function getReporter()
+    {
+        if ($this->coverage instanceof \PHP_CodeCoverage) {
+            return new CoverageReporter3($this->coverage);
+        } else {
```

Comment:
(minor) simply remove the `else {` here and `}` after since the previous block returns if it ran.